staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/swagger_util.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openapi
+
+import (
+	"crypto/sha512"
+	"encoding/json"
+	"fmt"
+
+	"github.com/go-openapi/spec"
+)
+
+// calcETag calculates etag for given swagger
+func calcETag(swagger *spec.Swagger) (string, error) {
```

Comment:
why does apiextensions-apiserver have logic about etags? This belongs into k8s.io/apiserver, to be shared everywhere. We shouldn't duplicate this kind of logic which defines an API.